crates/bevy_core_pipeline/src/taa/taa.wgsl

```diff
@@ -145,52 +130,30 @@ fn taa(@location(0) uv: vec2<f32>) -> Output {
-    let s_mr = sample_view_target(uv + vec2( texel_size.x,  0.0));
-    let s_bl = sample_view_target(uv + vec2(-texel_size.x, -texel_size.y));
-    let s_bm = sample_view_target(uv + vec2( 0.0,          -texel_size.y));
-    let s_br = sample_view_target(uv + vec2( texel_size.x, -texel_size.y));
-    let moment_1 = s_tl + s_tm + s_tr + s_ml + s_mm + s_mr + s_bl + s_bm + s_br;
-    let moment_2 = (s_tl * s_tl) + (s_tm * s_tm) + (s_tr * s_tr) + (s_ml * s_ml) + (s_mm * s_mm) + (s_mr * s_mr) + (s_bl * s_bl) + (s_bm * s_bm) + (s_br * s_br);
     let mean = moment_1 / 9.0;
     let variance = (moment_2 / 9.0) - (mean * mean);
     let std_deviation = sqrt(max(variance, vec3(0.0)));
-    history_color = RGB_to_YCoCg(history_color);
-    history_color = clip_towards_aabb_center(history_color, s_mm, mean - std_deviation, mean + std_deviation);
-    history_color = YCoCg_to_RGB(history_color);
-
-    // How confident we are that the history is representative of the current frame
-    var history_confidence = textureSample(history, nearest_sampler, uv).a;
-    let pixel_motion_vector = abs(closest_motion_vector) * texture_size;
-    if pixel_motion_vector.x < 0.01 && pixel_motion_vector.y < 0.01 {
-        // Increment when pixels are not moving
-        history_confidence += 10.0;
-    } else {
-        // Else reset
-        history_confidence = 1.0;
-    }
+    history_color = clip_towards_aabb_center(history_color, current_color, mean - std_deviation, mean + std_deviation);
 
-    // Blend current and past sample
-    // Use more of the history if we're confident in it (reduces noise when there is no motion)
-    // https://hhoppe.com/supersample.pdf, section 4.1
-    let current_color_factor = clamp(1.0 / history_confidence, MIN_HISTORY_BLEND_RATE, DEFAULT_HISTORY_BLEND_RATE);
-    current_color = mix(history_color, current_color, current_color_factor);
-#endif // #ifndef RESET
+    // Use more of the history if it's been visible for a few frames (reduces noise)
+    var accumulated_samples = textureSampleLevel(history, nearest_sampler, history_uv, 0.0).a;
+    // If the history_uv is pointing off-screen, reset accumulated sample count
+    accumulated_samples *= f32(all(saturate(history_uv) == history_uv));
+#ifdef RESET
+    accumulated_samples = 0.0;
+#endif
+    accumulated_samples = max(accumulated_samples + 1.0, 8.0);
 
+    // Blend current and past sample
+    current_color = mix(history_color, current_color, 1.0 / accumulated_samples);
 
     // Write output to history and view target
     var out: Output;
-#ifdef RESET
-    let history_confidence = 1.0 / MIN_HISTORY_BLEND_RATE;
-#endif
-    out.history = vec4(current_color, history_confidence);
+    out.history = vec4(current_color, accumulated_samples);
+    current_color = YCoCg_to_RGB(current_color);
 #ifdef TONEMAP
     current_color = reverse_tonemap(current_color);
 #endif
-    out.view_target = vec4(current_color, original_color.a);
+    out.view_target = vec4(current_color, 1.0);
```

Comment:
Won't this break the ability to use TAA with a transparent clear color for semitransparent windows?

Reply:
I suppose so. I don't think that's a big deal though. It's unclear what the alpha should even be, considering it's a blended value.

Reply:
Hmm it does mean that TAA cannot be used for augmented reality and some virtual production applications, unless we do the compositing with the video feed ourselves within bevy.

Can't we treat the alpha channel exactly like the other color channels and just blend it with a history, instead of special casing it?